Batch student status update

joshuayapwj98 opened the pull request.

joshuayapwj98 committed:
Add checks for Status and ClassCode

files edited:
- EditCommand.java: added status and classCode in isAnyFieldEdited() method to cater for the 2 fields.

joshuayapwj98 committed:
Add steps to approach batch student update

Edited files:
1. EditCommand.java: Added comments to document approach to the feature.

joshuayapwj98 committed:
Fix indentation

joshuayapwj98 committed:
Change location of documentation

lzf834 committed:
- Update UG for changes to new methods added - findstatus
and findclasscode
- Update own PPP to record new changes

lzf834 committed:
Checkstyle changes

Fenway17 committed:
Merge pull request #58 from e0544333/status-edit-command

Add checks for Status and ClassCode

lzf834 commented:
Checkstyle:
- Minor changes for the new file created, need new line at EOF in update_student_in_batches.

Fenway17 committed:
Merge pull request #63 from lzf834/v1.2_filter_command

Update UG and own PPP

Fenway17 commented:
Minor checkstyle errors detected by Java CI.

Fenway17 committed:
Update fenway17.md

lzf834 committed:
Merge pull request #64 from Fenway17/master

Updated personal PPP

lzf834 committed:
Implement JUnit Tests for Status and Status-related functions
- FindStatusTest
- FindStatusCommandParserTest
- StatusContainsKeywordsPredicateTest
- StatusTest

lzf834 committed:
Checkstyle changes for Status-related Tests

lzf834 committed:
CheckStyle Changes

whoisjunhong committed:
Merge pull request #69 from lzf834/v1.2_filter_command

Implement JUnit Tests for Status and Status-related functions

whoisjunhong committed:
Add filtered list by ClassCode and validation of student

whoisjunhong commented:
Closed since changes have been made and updated in #70 and 71

Fenway17 committed:
Merge pull request #70 from whoisjunhong/v1.2_batch_update_info

Add filtered list by ClassCode and validation of student

joshuayapwj98 committed:
Auto stash before merge of "master" and "origin/master"

joshuayapwj98 committed:
Merge remote-tracking branch 'origin/batch-student-status-update' into batch-student-status-update

louisdavinlie commented on src/main/java/seedu/address/logic/commands/EditCommand.java:
```diff
@@ -88,6 +91,24 @@ public CommandResult execute(Model model) throws CommandException {
 
         model.setPerson(personToEdit, editedPerson);
+
+        if (personToEdit.getStatus().toString().equals(Status.NEGATIVE)
+                && editedPerson.getStatus().toString().equals(Status.POSITIVE)) {
+
+            List<Person> filteredByClassCodeList = studentList.stream()
+                    .filter(student -> student.getClassCode().toString().equals(editedPerson.getClassCode().toString())
+                            && !student.isSamePerson(editedPerson))
+                    .collect(Collectors.toList());
+
+            for (int i = 0; i < filteredByClassCodeList.size(); i++) {
+                Person currentPerson = filteredByClassCodeList.get(i);
+                EditPersonDescriptor tempDescriptor = new EditPersonDescriptor();
+                tempDescriptor.setStatus(new Status(Status.CLOSE_CONTACT));
+                Person editedPersonStatus = createEditedPerson(currentPerson, tempDescriptor);
+                model.setPerson(currentPerson, editedPersonStatus);
+            }
+        }
```
Should we also handle the case where a student's status has changed from positive to negative? When a student has recovered, their classmates' status should be negative instead of close contact if there is no more positive student in the class.

louisdavinlie commented:
Looks good!

lzf834 commented:
Might need to consider an edge case where already Positive student should not be changed to close-contact upon update of a different student
